azurelinuxagent/common/utils/cryptutil.py

```diff
@@ -47,34 +47,30 @@ def gen_transport_cert(self, prv_file, crt_file):
                "-out {2}").format(self.openssl_cmd, prv_file, crt_file)
         rc = shellutil.run(cmd)
         if rc != 0:
-            logger.error("Failed to create {0} and {1} certificates".format(
-                prv_file, crt_file))
+            logger.error("Failed to create {0} and {1} certificates".format(prv_file, crt_file))
 
     def get_pubkey_from_prv(self, file_name):
         if not os.path.exists(file_name):
             raise IOError(errno.ENOENT, "File not found", file_name)
         else:
-            cmd = "{0} pkey -in {1} -pubout 2>/dev/null".format(self.openssl_cmd,
-                                                                file_name)
-            pub = shellutil.run_get_output(cmd)[1]
+            cmd = "{0} pkey -in {1} -pubout".format(self.openssl_cmd, file_name)
+            pub = shellutil.run_command(cmd.split(" "))
```

Comment:
Any particular reason for not using `shlex.split()` instead of `cmd.split(" ")`?

Reply:
Just wasn't aware of it. Thanks, will use it!